specification/evaluation-context/evaluation-context.md

```diff
@@ -1,5 +1,53 @@
 # Evaluation Context
 
+**Status**: [Experimental](../README.md#document-statuses)
+
 ## Overview
 
-// TODO: this is a stub.
+The `evaluation context` provides ambient information for the purposes of flag evaluation. Contextual data may be used as the basis for targeting, including rule-based evaluation, overrides for specific subjects, or fractional flag evaluation.
+
+### Fields
+
+See the table below for a summary of pre-defined fields.
+
+| Name          | Type   | Purpose                             | Example(s)                                                 |
+| ------------- | ------ | ----------------------------------- | ---------------------------------------------------------- |
+| targeting key | string | Subject identifier                  | "user@domain.com", "cb638b68-db72-11ec-9d64-0242ac120002"  |
+| ip            | string | Client IP address (IPv4 or IPv6)    | "216.7.132.218", "2001:0db8:85a3:0000:0000:8a2e:0370:7334" |
```

Comment:
Why do we need to have the client spectify this? It seems like something the API layer could (should?) automatically add via `gethostaddr` or similar. It's very odd to have a client go "this is my IP". Often the sever can also obtain it directly if needed from the host addr or proxy header info. I don't see any benefit in including it here (I'm happy to be educated)

Reply:
Something like `gethostaddr` would return the server IP, not the IP address of the user making the request. While it may be possible to automate collecting the client IP in some situations, I would argue that we don't want to. It should be up to the developer/business to decide if that information is required. It also becomes a bit more tricky if the server is behind proxies because that information may only be available via a header such as `X-Forwarded-For`

Reply:
>Something like gethostaddr would return the server IP, not the IP address of the user making the request. 

For sever-side (backend) flag evaluation, that is true. But for browser SDKs, the OpenFeature APIs are called in the javascript in the end-user's browser, not the server. Thus the IP specified in this evaluation context would/could be looked up locally by the browser. The browser code accessing the OpenFeature SDK won't know about any proxies anyway, and will probably see the local IP address as 192.168.1.1 on the local subnet. I really don't see how setting the `ip` in the evaluation context helps anyone here.